Automattic/wp-calypso
# Check if ETK is disabled using a site option

This PR reimplements the "is Editing Toolkit disabled on an Atomic site" checks from #47569.

It's a part of the `InlineHelp` extraction experiment in #55189 and removes one of Redux dependencies (`state.plugins`) of the `SiteSelector` component.

Companion PR for the server D65864-code patch that updates the `is_coming_soon` site option calculation and adds a new `editing_toolkit_is_active` site option.

When displaying the "Coming Soon" badge on a site, we use the updated `is_coming_soon` site option, where the server itself does the editing-toolkit-disabled check and sends `false` value when ETK is indeed disabled. Thanks to that, we don't need to do any additional check (and plugin info fetching) in Calypso. The `is_coming_soon` option value is the entire truth.

In Settings > General for an Atomic site, we want to disable the "Coming Soon" privacy option when ETK is disabled. To do that, we use the new `editing_toolkit_is_active` site option. Again, only the `state.sites` Redux subtree is needed for that, and we don't need to depend on `state.plugins`.

**How to test:**
- apply the D65864-code patch to your sandbox and sandbox `public-api.wordpress.com`
- start Calypso with this PR applied
- switch your Atomic site into a "Coming Soon" mode and then disable ETK
- in the site selector, verify that the Atomic site _doesn't_ have the "Coming Soon" badge
- in Settings > General, verify that the "Coming Soon" privacy option is not available when ETK is disabled
- after enabling ETK again, verify that "Coming Soon" badge is back and that the "Coming Soon" setting is available in settings
